### PR TITLE
Add Fleet 

### PR DESCRIPTION
**Pull request by aag6z**
Several commits to add fleet to automation

### CODE REVIEW

**Review by sunilemanjee — COMMENTED** [review]
@aag6z  why change from using var to locals?

File: gcp/eck/data-sources.tf [hunk 1]
```diff
@@ -16,11 +25,11 @@ data "kubectl_path_documents" "es" {
         cold_pod_ES_JAVA_OPTS = var.cold_pod_ES_JAVA_OPTS
         frozen_pod_ES_JAVA_OPTS = var.frozen_pod_ES_JAVA_OPTS
         ml_pod_ES_JAVA_OPTS = var.ml_pod_ES_JAVA_OPTS
-        master_pod_roles = var.master_pod_roles
-        hot_pod_roles = var.hot_pod_roles
-        warm_pod_roles = var.warm_pod_roles
-        cold_pod_roles = var.cold_pod_roles
-        frozen_pod_roles = var.frozen_pod_roles
+        master_pod_roles = local.master_roles
```
Comment (sunileman) [inline gcp/eck/data-sources.tf]: @aag6z  not sure why the change from using var to locals is required here.

File: gcp/eck/data-sources.tf [hunk 2]
```diff
@@ -5,6 +5,15 @@ data "terraform_remote_state" "k8s" {
     path = "../gke/terraform.tfstate"
   }
 }
+
+locals {
+    master_roles = var.kibana_fleet_enabled ? "${var.master_pod_roles}, remote_cluster_client" : var.master_pod_roles
```
Comment (sunileman) [inline gcp/eck/data-sources.tf]: @aag6z  wouldn't it be easier to just add remove cluster client to all roles instead of check if fleet is enable? Any downside to having remote_cluster_client on all roles?
